libs/ebpfnetsh/programs.cpp

```diff
@@ -176,93 +138,44 @@ handle_ebpf_add_program(
         return status;
     }
 
-    const char* error_message = nullptr;
-    program_state_t program(filename, section);
-    uint32_t count_of_map_handles = _countof(program.map_handles);
-    status = ebpf_api_load_program(
-        filename.c_str(),
-        section.c_str(),
-        execution,
-        &program.program_handle,
-        &count_of_map_handles,
-        program.map_handles,
-        &error_message);
-    if (status != ERROR_SUCCESS) {
-        if (error_message != nullptr) {
-            std::cerr << error_message << std::endl;
-        } else {
-            std::cerr << "error " << status << ": could not load program" << std::endl;
-        }
-        ebpf_free_string(error_message);
-        program.clean();
+    struct bpf_object* object;
+    int program_fd;
+    PCSTR error_message;
+    ebpf_result_t result = ebpf_program_load(
+        filename.c_str(), program_type, attach_type, EBPF_EXECUTION_ANY, &object, &program_fd, &error_message);
+    if (result != EBPF_SUCCESS) {
+        std::cerr << "error " << result << ": could not load program" << std::endl;
+        std::cerr << error_message << std::endl;
```

Comment:
We are not freeing `error_message` in case `ebpf_program_load` fails.

Reply:
updated